test/integration/apiserver/admissionwebhook/match_conditions_test.go

```diff
@@ -676,6 +565,150 @@ func Test_MatchConditions(t *testing.T) {
 	}
 }
 
+func TestMatchConditions_validation(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AdmissionWebhookMatchConditions, true)()
+
+	server := apiservertesting.StartTestServerOrDie(t, nil, []string{
+		"--disable-admission-plugins=ServiceAccount",
+	}, framework.SharedEtcd())
+	defer server.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(server.ClientConfig)
+
+	testcases := []struct {
+		name            string
+		matchConditions []admissionregistrationv1.MatchCondition
+		expectError     bool
```

Comment:
I don't recall why `TestMatchConditions` didn't check the contents of the error, but should we be doing that here? It seems more do-able for validation errors